sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java

```diff
@@ -67,6 +67,21 @@
   private static final BiMap<SqlTypeName, FieldType> CALCITE_TO_BEAM_TYPE_MAPPING =
       BEAM_TO_CALCITE_TYPE_MAPPING.inverse();
 
+  // The list of field type names used in SQL as Beam field types.
+  public static final FieldType TINY_INT = toFieldType(SqlTypeName.TINYINT);
+  public static final FieldType SMALL_INT = toFieldType(SqlTypeName.SMALLINT);
+  public static final FieldType INTEGER = toFieldType(SqlTypeName.INTEGER);
+  public static final FieldType BIG_INT = toFieldType(SqlTypeName.BIGINT);
+  public static final FieldType FLOAT = toFieldType(SqlTypeName.FLOAT);
+  public static final FieldType DOUBLE = toFieldType(SqlTypeName.DOUBLE);
+  public static final FieldType DECIMAL = toFieldType(SqlTypeName.DECIMAL);
+  public static final FieldType BOOLEAN = toFieldType(SqlTypeName.BOOLEAN);
+  public static final FieldType CHAR = toFieldType(SqlTypeName.CHAR);
+  public static final FieldType VARCHAR = toFieldType(SqlTypeName.VARCHAR);
```

Comment:
The main reason I was reluctant to delete the `RowSqlTypes` is because Calcite has different type names than our schema, and it can be confusing when mixed together, contrived example:

```
Schema
  .builder()

  .addInt32Field("f_int")
  .addField(Field.of("f_alsoInt", CalciteUtils.INTEGER))

  .addDateTimeField("f_dateTime")
  .addField(Field.of("f_timestamp", CalciteUtils.TIMESTAMP))

  .addStringField("f_str")
  .addField(Field.of("f_char", CalciteUtils.CHAR)
  .addField(Field.of("f_varchar", CalciteUtils.VARCHAR)

  .build()
```

Reply:
Or when creating a schema and then using different types in the queries

Reply:
This looks fine, we should merge this.

I think we will move towards the more normal way as soon as we start integrating more type systems. It then will become clearer what specific abstractions we need and how to tie them together.